### PR TITLE
159799469 configure messaging for service bus

### PR DESCRIPTION
**Pull request by jhc5000**

### CODE REVIEW

File: src/ServiceHub.Room.Service/Controllers/RoomController.cs [hunk 1]
```diff
@@ -14,11 +17,23 @@ public class RoomController : BaseController
     {
         private readonly IRoomsRepository _context;
 
+        public string ConnectionString { get; }
+        public string QueueName { get; }
+        static IQueueClient struggleBusser;
```
Comment (fredbelotte) [inline src/ServiceHub.Room.Service/Controllers/RoomController.cs]: Code Smell Code Smell: Remove the 'struggleBusser' field and declare it as a local variable in the relevant methods. ([csharpsquid:S1450](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS1450))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=servicehub%3Aroom&pullRequest=36&issues=AWVJqlUolRPS_ea-SUaP&open=AWVJqlUolRPS_ea-SUaP)

[ISSUE_KEY]: AWVJqlUolRPS_ea-SUaP

File: src/ServiceHub.Room.Service/Controllers/RoomController.cs [hunk 2]
```diff
@@ -14,11 +17,23 @@ public class RoomController : BaseController
...
+        static IQueueClient struggleBusser2;
+
         public RoomController(ILoggerFactory loggerFactory, IRoomsRepository repo) : base(loggerFactory)
         {
             _context = repo;
+
+            // ConnectionString = Configuration.GetConnectionString("ServiceBus");
```
Comment (fredbelotte) [inline src/ServiceHub.Room.Service/Controllers/RoomController.cs]: Code Smell Code Smell: Remove this commented out code. ([csharpsquid:S125](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS125))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=servicehub%3Aroom&pullRequest=36&issues=AWVJqlUolRPS_ea-SUaT&open=AWVJqlUolRPS_ea-SUaT)

[ISSUE_KEY]: AWVJqlUolRPS_ea-SUaT

File: src/ServiceHub.Room.Service/Controllers/RoomController.cs [hunk 3]
```diff
@@ -30,22 +45,48 @@ public async Task<IActionResult> Get()
...
-            var rooms = ModelMapper.ContextToLibrary(ctxRooms);
+                struggleBusser2 = new QueueClient("Endpoint=sb://johncservice.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=LsBw+qYLdOjz5Ag1ia+z8QmNoFOvPh2oUZ8tCGBqGE8=", "bpjtest1");
 
-            if (rooms.Count == 0)
+                if (rooms.Count == 0)
+                {
+                    logger.LogError("Rooms pulled from database cannot be converted to a room model or were invalid.", null);
+                    return StatusCode(500);
+                }
+                string jsonobj = JsonConvert.SerializeObject(rooms);
```
Comment (fredbelotte) [inline src/ServiceHub.Room.Service/Controllers/RoomController.cs]: Code Smell Code Smell: Remove this unused 'jsonobj' local variable. ([csharpsquid:S1481](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS1481))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=servicehub%3Aroom&pullRequest=36&issues=AWVJqlUolRPS_ea-SUaR&open=AWVJqlUolRPS_ea-SUaR)

[ISSUE_KEY]: AWVJqlUolRPS_ea-SUaR

File: src/ServiceHub.Room.Service/Controllers/RoomController.cs [hunk 4]
```diff
@@ -30,22 +45,48 @@ public async Task<IActionResult> Get()
...
+                string messageBody = "msg";
+                var message = new Message(Encoding.UTF8.GetBytes(messageBody));
+                //await struggleBusser.SendAsync(message);
```
Comment (fredbelotte) [inline src/ServiceHub.Room.Service/Controllers/RoomController.cs]: Code Smell Code Smell: Remove this commented out code. ([csharpsquid:S125](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS125))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=servicehub%3Aroom&pullRequest=36&issues=AWVJqlUolRPS_ea-SUaU&open=AWVJqlUolRPS_ea-SUaU)

[ISSUE_KEY]: AWVJqlUolRPS_ea-SUaU

File: src/ServiceHub.Room.Service/Controllers/RoomController.cs [hunk 5]
```diff
@@ -30,22 +45,48 @@ public async Task<IActionResult> Get()
...
+                logger.LogError(e, "No rooms were found in the database. Failed to get rooms. ", null);
                 return StatusCode(500);
             }
-
-            return Ok(rooms);
+            
+            
+
+            //try
+            //{
```
Comment (fredbelotte) [inline src/ServiceHub.Room.Service/Controllers/RoomController.cs]: Code Smell Code Smell: Remove this commented out code. ([csharpsquid:S125](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS125))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=servicehub%3Aroom&pullRequest=36&issues=AWVJqlUolRPS_ea-SUaV&open=AWVJqlUolRPS_ea-SUaV)

[ISSUE_KEY]: AWVJqlUolRPS_ea-SUaV

File: src/ServiceHub.Room.Service/Controllers/RoomController.cs [hunk 6]
```diff
@@ -30,22 +45,48 @@ public async Task<IActionResult> Get()
...
-            }
-            catch (Exception e)
-            {
-                logger.LogError(e, "No rooms were found in the database.", null);
-                return StatusCode(500);
-            }
+                var rooms = ModelMapper.ContextToLibrary(ctxRooms);
 
-            var rooms = ModelMapper.ContextToLibrary(ctxRooms);
+                struggleBusser2 = new QueueClient("Endpoint=sb://johncservice.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=LsBw+qYLdOjz5Ag1ia+z8QmNoFOvPh2oUZ8tCGBqGE8=", "bpjtest1");
```
Comment (fredbelotte) [inline src/ServiceHub.Room.Service/Controllers/RoomController.cs]: Code Smell Code Smell: Make the enclosing instance method 'static' or remove this set on the 'static' field. ([csharpsquid:S2696](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS2696))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=servicehub%3Aroom&pullRequest=36&issues=AWVJqlUolRPS_ea-SUaS&open=AWVJqlUolRPS_ea-SUaS)

[ISSUE_KEY]: AWVJqlUolRPS_ea-SUaS

File: src/ServiceHub.Room.Service/Controllers/RoomController.cs [hunk 7]
```diff
@@ -14,11 +17,23 @@ public class RoomController : BaseController
...
+        static IQueueClient struggleBusser;
+        static IQueueClient struggleBusser2;
+
         public RoomController(ILoggerFactory loggerFactory, IRoomsRepository repo) : base(loggerFactory)
         {
             _context = repo;
+
+            // ConnectionString = Configuration.GetConnectionString("ServiceBus");
+            QueueName = "usertohs";
+            struggleBusser = new QueueClient("Endpoint=sb://housingbus.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=eqzwmaQMeEBHcfA6AQjwz5GHHwo78BPQpDgfdukjOZM=", QueueName);
```
Comment (fredbelotte) [inline src/ServiceHub.Room.Service/Controllers/RoomController.cs]: Code Smell Code Smell: Remove this assignment of 'struggleBusser' or initialize it statically. ([csharpsquid:S3010](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS3010))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=servicehub%3Aroom&pullRequest=36&issues=AWVJqlUolRPS_ea-SUaQ&open=AWVJqlUolRPS_ea-SUaQ)

[ISSUE_KEY]: AWVJqlUolRPS_ea-SUaQ

File: src/ServiceHub.Room.Service/Controllers/RoomController.cs [hunk 8]
```diff
@@ -14,11 +17,23 @@ public class RoomController : BaseController
     {
         private readonly IRoomsRepository _context;
 
+        public string ConnectionString { get; }
+        public string QueueName { get; }
+        static IQueueClient struggleBusser;
+        static IQueueClient struggleBusser2;
```
Comment (fredbelotte) [inline src/ServiceHub.Room.Service/Controllers/RoomController.cs]: Code Smell Code Smell: Remove the 'struggleBusser2' field and declare it as a local variable in the relevant methods. ([csharpsquid:S1450](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS1450))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=servicehub%3Aroom&pullRequest=36&issues=AWVJqlUolRPS_ea-SUaO&open=AWVJqlUolRPS_ea-SUaO)

[ISSUE_KEY]: AWVJqlUolRPS_ea-SUaO

**Comment by Hoagithor** [comment]
1 code smell will remain, referencing _queues in RoomController. This is due to its lack of use, and will be resolved when queue use and implementation is begun.

File: src/ServiceHub.Room.Service/Controllers/RoomController.cs [hunk 9]
```diff
@@ -13,12 +15,16 @@ namespace ServiceHub.Room.Service.Controllers
     public class RoomController : BaseController
     {
         private readonly IRoomsRepository _context;
+        private readonly IQueueFactory _queues;
```
Comment (fredbelotte) [inline src/ServiceHub.Room.Service/Controllers/RoomController.cs]: Code Smell Code Smell: Remove the '_queues' field and declare it as a local variable in the relevant methods. ([csharpsquid:S1450](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS1450))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=servicehub%3Aroom&pullRequest=36&issues=AWVZKda0lRPS_ea-f-V2&open=AWVZKda0lRPS_ea-f-V2)

[ISSUE_KEY]: AWVZKda0lRPS_ea-f-V2

Comment (fredbelotte) [inline src/ServiceHub.Room.Service/Controllers/RoomController.cs]: Code Smell Code Smell: Remove the '_queues' field and declare it as a local variable in the relevant methods. ([csharpsquid:S1450](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS1450))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=servicehub%3Aroom&pullRequest=36&issues=AWVZKda0lRPS_ea-f-V2&open=AWVZKda0lRPS_ea-f-V2)

[ISSUE_KEY]: AWVZKda0lRPS_ea-f-V2

File: src/ServiceHub.Room.Service/Controllers/RoomController.cs [hunk 10]
```diff
@@ -13,12 +15,16 @@ namespace ServiceHub.Room.Service.Controllers
...
 
-        public RoomController(ILoggerFactory loggerFactory, IRoomsRepository repo) : base(loggerFactory)
+        public RoomController(ILoggerFactory loggerFactory, IRoomsRepository repo, IQueueFactory queues) : base(loggerFactory)
         {
             _context = repo;
+           _queues = queues;
         }
 
+
+
```
Comment (escalonn) [inline src/ServiceHub.Room.Service/Controllers/RoomController.cs]: unneeded spacing and bad indentation

File: src/ServiceHub.Room.Service/Startup.cs [hunk 11]
```diff
@@ -30,6 +31,9 @@ public void ConfigureServices(IServiceCollection services)
 
             services.AddMvc();
 
+            string busConnection = Configuration.GetConnectionString("ServiceBus");
+            services.AddScoped<IQueueFactory>(sp => new QueueFactory(busConnection));
```
Comment (escalonn) [inline src/ServiceHub.Room.Service/Startup.cs]: this should be a singleton.

File: src/ServiceHub.Room.Service/Controllers/RoomController.cs [hunk 12]
```diff
@@ -13,10 +15,12 @@ namespace ServiceHub.Room.Service.Controllers
     public class RoomController : BaseController
     {
         private readonly IRoomsRepository _context;
+        private readonly IQueueFactory _queues;
```
Comment (fredbelotte) [inline src/ServiceHub.Room.Service/Controllers/RoomController.cs]: Code Smell Code Smell: Remove the '_queues' field and declare it as a local variable in the relevant methods. ([csharpsquid:S1450](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS1450))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=servicehub%3Aroom&pullRequest=36&issues=AWVZKda0lRPS_ea-f-V2&open=AWVZKda0lRPS_ea-f-V2)

[ISSUE_KEY]: AWVZKda0lRPS_ea-f-V2

Comment (Hoagithor) [reply src/ServiceHub.Room.Service/Controllers/RoomController.cs]: As mentioned above, this is unavoidable as we have this up now for use during implementation.

**Pull request merged**